src/hooks/login/useWalletConnectLogin.ts

```diff
@@ -123,6 +123,11 @@ export const useWalletConnectLogin = ({
       }
 
       const address = await provider.getAddress();
+      if (!address) {
```

Comment:
The logic looks slightly different : here if there is no address there is a return, in the others there's an if for the address. Maybe we should extract that logic into a function that accepts the address, loginMethod, callbackRoute and onLoginRedirect and do all that for all providers?

Reply:
In the other cases there is a finally {} block that stops the loading indicator. Here is not the case and we can stop the execution early. I am checking if I can extract some common logic